### PR TITLE
Added some logging information to the totality checker.

### PR DESCRIPTION
**Pull request by jfdm**
This will hopefully make debugging of the totality/termination checker that little bit easier, especially given #645.

I need someone to check the levels, as I think I have sett them to high...

### CODE REVIEW

File: src/Core/Termination.idr [hunk 1]
```diff
@@ -362,7 +363,7 @@ mutual
       = do Just gdef <- lookupCtxtExact fn_in (gamma defs)
                 | Nothing => throw (UndefinedName fc fn_in)
            let fn = fullname gdef
-           log "termination" 10 $ "Looking under " ++ show fn
+           log "termination.termination.sizechange" 10 $ "Looking under " ++ show fn
```
Comment (gallais) [inline src/Core/Termination.idr:366]: `termination.termination`?

Comment (jfdm) [reply src/Core/Termination.idr:366]: i was carrying on from the existing topic! Maybe we should change the lead topic to be `totality`?

Comment (gallais) [reply src/Core/Termination.idr:366]: I see. I think `totality` would indeed be clearer. Or just not having `termination` and
`positivity` under the same heading (we can always use multiple topics anyways).

Comment (jfdm) [reply src/Core/Termination.idr]: TBH I think what is also happening is that I assume that topics are presented hierarchically i.e. "grandparent.parent.child". This is probably where our mental models differ. Regardless the change has been made.

File: src/Core/Termination.idr [hunk 2]
```diff
@@ -410,7 +410,7 @@ export
 calculateSizeChange : {auto c : Ref Ctxt Defs} ->
                       FC -> Name -> Core (List SCCall)
 calculateSizeChange loc n
-    = do log "termination.sizechange" 5 $ "Calculating Size Change: " ++ show n
+    = do log "totality.sizechange" 5 $ "Calculating Size Change: " ++ show n
```
Comment (gallais) [inline src/Core/Termination.idr:413]: This should probably be `totality.termination.sizechange` for uniformity
or is it an unrelated pass (in which case a different name would be clearer)?

Comment (jfdm) [reply src/Core/Termination.idr:413]: I think a different name is required, same for the one for guarded. They don't appear as subcalls to existing functions in the module and are both exported.

Comment (gallais) [reply src/Core/Termination.idr:413]: Tbh I tend to think of these in terms of topics, not call graphs.

Comment (jfdm) [reply src/Core/Termination.idr]: I've made the appropriate changes, and think that his is good to go.

File: src/Core/Termination.idr [hunk 3]
```diff
@@ -43,7 +43,7 @@ export
 checkIfGuarded : {auto c : Ref Ctxt Defs} ->
                  FC -> Name -> Core ()
 checkIfGuarded fc n
-    = do log "totality.guarded" 6 $ "Check if Guarded: " ++ show n
+    = do log "totality..termination.guarded" 6 $ "Check if Guarded: " ++ show n
```
Comment (gallais) [inline src/Core/Termination.idr:46]: two dots

Comment (jfdm) [reply src/Core/Termination.idr:46]: urgh

**Pull request merged**